packages/main/test-integration/node/cache-content.test.ts

```diff
@@ -0,0 +1,266 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import { GoogleGenerativeAI } from "../..";
+import { GoogleAICacheManager } from "../../server"; //= require("@google/generative-ai/server");
+
+use(chaiAsPromised);
+
+/**
+ * Integration tests against live backend.
+ */
+
+describe("cacheContent", function () {
```

Comment:
We only have positive test cases. Is it worth testing cases where we expect errors? For example, testing that we get an error if we pass an invalid cached content name with blank spaces, special characters, empty string, etc...
Example in Python SDK: https://github.com/google-gemini/generative-ai-python/pull/355/files#diff-e57945cd50529ced9f0b8aa12022ab5140a6c706d74ecfd05280778f707d3d36R161-R177

Reply:
Discussed this offline. There are some unit tests already for negative scenarios where we test the SDK behavior itself. I will work to add some purposely failing integration tests, too, but that shouldn't block this release.